src/Hazelcast.Net.Tests/Remote/UnisocketTests.cs

```diff
@@ -0,0 +1,108 @@
+                 {
+                     HConsole.WriteLine(this, args.State);
+                     if (args.State == ClientState.Connected || args.State == ClientState.Shutdown)
+                         eventCount++;
+                 }));
+             });
+
+            await client.DisposeAsync();
+
+            Assert.AreEqual(2, eventCount);
+        }
+
+        [Test]
+        public async Task TestDistributedObjectEventsWithDummyClient()
+        {            
+            var memberA = await AddMember();
+            var memberB = await AddMember();
+
+            int eventCount = 0;
+
+            var client = await CreateAndStartClientAsync(opt =>
+            {
+                opt.Networking.SmartRouting = false;                                
+            });
+
+            var map = await client.GetMapAsync<int, int>("myMap");
+            await map.SubscribeAsync(events => events.EntryAdded((sender, args) => { eventCount++; }));
+
+            await ShutdownMember(Guid.Parse(memberA.Uuid));
+            memberA = await AddMember();
+            
+            await AssertEx.SucceedsEventually(async () =>
+            {
+                var script = "instance_0.getMap(\"myMap\").put(1,1)";
+                var response = await RcClient.ExecuteOnControllerAsync(RcCluster.Id, script, Hazelcast.Testing.Remote.Lang.JAVASCRIPT);
+                Assert.GreaterOrEqual(1, eventCount);
+            }, 10_000, 500);
+        }
+
+        /// <summary>
+        /// Port of testMemberConnectionOrder
+        /// </summary>
+        /// <returns></returns>
+        [Test]
+        public async Task TestClientConnectsToOneMember()
+        {
+            var memberA = await AddMember();
+            var memberB = await AddMember();
+
+            var client = await CreateAndStartClientAsync(opt =>
+            {
+                opt.Networking.SmartRouting = false;
+                opt.Networking.Addresses.Clear();
+                opt.Networking.Addresses.Add(memberA.Host + ":" + memberA.Port);
+                opt.Networking.Addresses.Add(memberB.Host + ":" + memberB.Port);
+            });
+
+            var script = "result = instance_0.getClientService().getConnectedClients().size().toString();";
+            var response = await RcClient.ExecuteOnControllerAsync(RcCluster.Id, script, Hazelcast.Testing.Remote.Lang.JAVASCRIPT);
+
+            var countOfClients = int.Parse(Encoding.UTF8.GetString(response.Result));
+
+            // instance1 shouldn't have any clients due to unisocket mode
+            script = "result = instance_1.getClientService().getConnectedClients().size().toString();";
+            response = await RcClient.ExecuteOnControllerAsync(RcCluster.Id, script, Hazelcast.Testing.Remote.Lang.JAVASCRIPT);
+            countOfClients += int.Parse(Encoding.UTF8.GetString(response.Result));
+
+            //sum should be 1 since don't know which member is connected to
+            Assert.AreEqual(1, countOfClients);
```

Comment:
Can you also map put 100 entries and check the count again just in case there may be a late opening connection?

Reply:
What do you mean by late opening?

Reply:
@emreyigit  Will update the test.